cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/authenticator/OAuthUserAuthenticator.java

```diff
@@ -0,0 +1,61 @@
+package org.jasig.cas.support.oauth.authenticator;
+
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.authentication.AuthenticationManager;
+import org.jasig.cas.authentication.AuthenticationException;
+import org.jasig.cas.authentication.AuthenticationTransaction;
+import org.jasig.cas.authentication.UsernamePasswordCredential;
+import org.jasig.cas.authentication.principal.Principal;
+import org.jasig.cas.support.oauth.profile.OAuthUserProfile;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+import org.pac4j.http.credentials.authenticator.UsernamePasswordAuthenticator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Authenticator for user credentials authentication.
+ *
+ * @author Jerome Leleu
+ * @since 4.3.0
+ */
+@Component("oAuthUserAuthenticator")
+public class OAuthUserAuthenticator implements UsernamePasswordAuthenticator {
+
+    @NotNull
+    @Autowired
+    @Qualifier("authenticationManager")
+    private AuthenticationManager authenticationManager;
+
+    @Override
+    public void validate(final UsernamePasswordCredentials credentials) {
+        final UsernamePasswordCredential casCredential = new UsernamePasswordCredential(credentials.getUsername(),
+                credentials.getPassword());
+        try {
+            final Authentication authentication = authenticationManager.authenticate(AuthenticationTransaction.wrap(casCredential));
+            final Principal principal = authentication.getPrincipal();
+
+            final OAuthUserProfile profile = new OAuthUserProfile();
```

Comment:
All theses classes (authenticators and profiles) are not meant to be extensible. I will make them `final` for now and we'll see if any extension request comes in.